### PR TITLE
fix(amazon/deploy): Edit deployment cluster button did not work

### PR DESCRIPTION
**Pull request by jervi**
Nothing happened when pressing the edit icon, apart for the following console error being printed: "Error: One of component or template or templateUrl options is required".
You should of course just enter the correct configuration values immediately, but some people are not perfect and do mistakes. This commit will make those people happy, because the edit button once again works.

### CODE REVIEW

**Comment by christopherthielen** [comment]
Thanks, I was just looking into this too!  FYI there are at least a couple of other defects in this refactor that I'm hoping to tackle soon.

File: app/scripts/modules/core/src/pipeline/config/stages/deploy/deployStage.js [hunk 1]
```diff
@@ -162,34 +162,38 @@ module.exports = angular
...
+                resolve: {
+                  title: () => title,
+                  application: () => application,
+                  serverGroupCommand: () => command,
+                },
+              })
+              .result.then(handleResult)
+              .catch(() => {});
+          }
+        });
```
Comment (christopherthielen) [inline app/scripts/modules/core/src/pipeline/config/stages/deploy/deployStage.js]: move `handleResult` to here, right?

File: app/scripts/modules/core/src/pipeline/config/stages/deploy/deployStage.js [hunk 2]
```diff
@@ -162,34 +162,38 @@ module.exports = angular
...
+                templateUrl: providerConfig.serverGroup.cloneServerGroupTemplateUrl,
+                controller: `${providerConfig.serverGroup.cloneServerGroupController} as ctrl`,
+                size: 'lg',
+                resolve: {
+                  title: () => title,
+                  application: () => application,
+                  serverGroupCommand: () => command,
+                },
+              })
+              .result.then(handleResult)
```
[the remark this answers is not in the document]
Comment (jervi) [reply app/scripts/modules/core/src/pipeline/config/stages/deploy/deployStage.js]: You're probably right. Fixed, and I also did the same for the `addCluster` function. Can you confirm that this is correct?

[the remark this answers is not in the document]
Comment (jervi) [reply app/scripts/modules/core/src/pipeline/config/stages/deploy/deployStage.js]: I'm unsure of the return promise of the react blocks.

**Comment by christopherthielen** [comment]
lgtm, thanks!

**Pull request merged**